docs/source/operations/conversions/topocentric.rst

```diff
@@ -0,0 +1,104 @@
+.. _topocentric:
+
+================================================================================
+Geocentric to topocentric conversion
+================================================================================
+
+.. versionadded:: 8.0.0
+
+Convert geocentric coordinates to topocentric coordinates (in the forward path).
+
++---------------------+--------------------------------------------------------+
+| **Alias**           | topocentric                                            |
++---------------------+--------------------------------------------------------+
+| **Domain**          | 3D                                                     |
++---------------------+--------------------------------------------------------+
+| **Input type**      | Geocentric cartesian coordinates                       |
++---------------------+--------------------------------------------------------+
+| **Output type**     | Topocentric cartesian coordinates                      |
++---------------------+--------------------------------------------------------+
+
+This operation converts geocentric coordinate values (X, Y, Z) to topocentric
+(E/East, N/North, U/Up) values. This is also sometimes known as the ECEF (Earth
+Centered Earth Fixed) to ENU conversion.
+
+Topocentric coordinates are expressed in a frame whose East and North axis form
+a local tangent plane to the Earth's ellipsoidal surface fixed to a specific
+location (the topocentric origin), and the Up axis points upwards along the
+normal to that plane.
+
+.. image:: ./images/ECEF_ENU_Longitude_Latitude_relationships.png
+   :align: center
+   :scale: 100%
+   :alt:   ENU coordinate frame
+
+..
+  Source : https://en.wikipedia.org/wiki/Local_tangent_plane_coordinates#/media/File:ECEF_ENU_Longitude_Latitude_relationships.svg
+  Public domain
+..
+
+The topocentric origin is a required parameter of the conversion, and can be
+expressed either as geocentric coordinates (``X_0``, ``Y_0`` and ``Z_0``) or
+as geographic coordinates (``lat_0``, ``lon_0``, ``h_0``).
+
+When conversion between geographic and topocentric coordinates is desired, the
+topocentric conversion must be preceded by the :ref:`cart` conversion to
+perform the initial geographic to geocentric coordinates conversion.
```

Comment:
I most often see ENU-coordinates exressed in degrees (e.g. the model input for +proj=deformation which I believe to be similar to this) would that then be a matter of doing a pipeline like `+proj=pipeline +step ... +step +proj=topocentric ... +step +proj=cart +inv`? If that is the case, I think an example for that would be useful too.

Reply:
Hum, from my testing, it doesn't seem that using +proj=cart +inv would give the expected result. In defmodel_impl.hpp, there's the function
```
inline void DeltaEastingNorthingToLongLat(double cosphi, double de, double dn,
                                          double a, double b, double es,
                                          double &dlam, double &dphi) {
    const double oneMinuX = es * (1 - cosphi * cosphi);
    const double X = 1 - oneMinuX;
    const double sqrtX = sqrt(X);
    dlam = de * sqrtX / (a * cosphi);
    dphi = dn * a * sqrtX * X / (b * b);
}
```
and if doing for example "echo 4.99 55.1 73 | src/cct -d 3 +proj=pipeline +step +proj=affine +step +proj=cart +ellps=WGS84 +step +proj=topocentric +ellps=WGS84 +lon_0=5 +lat_0=55 +h_0=200", and injecting the resulting de, dn into it, it gives the expected dlon ~= -0.001 and dlat ~= 0.1. Whereas using +proj=cart +inv gives completely different and wrong results.

I'd suspect DeltaEastingNorthingToLongLat() to be an approximation only valid for small deltas. Maybe @ccrook can give some hints ?